odxtools/parameters/parameterwithdop.py

```diff
@@ -66,7 +66,10 @@ def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
 
         if self.dop_snref:
             ddds = diag_layer.diag_data_dictionary_spec
-            self._dop = odxrequire(ddds.all_data_object_properties.get(self.dop_snref))
+            if snref_shortname := ddds.all_data_object_properties.get(self.dop_snref):
```

Comment:
That `if` statment is equivalent to simply
```python
self._dop = ddds.all_data_object_properties.get(self.dop_snref)
```

that said, I think the `odxrequire()` should stay because the reference to the DOP must be resolved. If it isn't (most probably because the referenced DOP is not yet implemented, e.g., `DYNAMIC-ENDMARKER-FIELD`, one should use the non-strict mode for now. This is because quite a bit of code assumes that `ParameterWithDop` objects provide a DOP...)